code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -197,10 +203,8 @@ void setNetwork(Map<String, Object> inspect, Instance instance) {
         }
     }
 
-    @SuppressWarnings("unchecked")
-    void setImage(Map<String, Object> inspect, Instance instance) {
-        Map<String, Object> config = (Map<String, Object>)inspect.get(INSPECT_CONFIG);
-        String imageUuid = (String)config.get(INSPECT_IMAGE);
+    void setImage(ContainerEvent event, Instance instance) {
+        String imageUuid = event.getExternalFrom();
```

Comment:
docker events include the image that was used for the container and this value is identical to what is in the inspect. We don't have the inspect for a container why doing the the sync via ping, but we do have it, so it is more straightforward to set it on the event as opposed to trying to construct a "fake" inspect. Does that make sense @sonchang ?